examples/outlet-form-rerender/app/routes/users/$userId.tsx

```diff
@@ -0,0 +1,70 @@
+import type { LoaderFunction, MetaFunction } from "remix";
+import { useCatch } from "remix";
+import { json, useLoaderData } from "remix";
+import type { User as UserType } from "~/data.server";
+import { users } from "~/data.server";
+
+interface LoaderData {
+  user: UserType;
+}
+
+export const meta: MetaFunction = ({ data }) => {
+  if (!data) {
+    return { title: "User not found!" };
+  }
+  return { title: (data as LoaderData).user.name };
+};
+
+export const loader: LoaderFunction = async ({ params }) => {
+  const userId = params.userId;
+
+  const user = users.find(({ id }) => id === userId);
+
+  if (!user) {
+    throw json(null, { status: 404 });
+  }
+
+  return { user } as LoaderData;
+};
+export default function User() {
+  const { user } = useLoaderData<LoaderData>();
+
+  return (
+    /*
+     * Since we're setting the initial value of the input through
+     * defaultValue, React will initialise the input field with the
+     * initial value and leave the subsequent updates uncontrolled
+     *
+     * To rerender the input field with the updated value on the route
+     * transition, we have to inform React to unmount and remount the component
+     * which resets the initial values
+     *
+     * https://reactjs.org/docs/uncontrolled-components.html
+     * */
+    <form method="post" key={user.id}>
```

Comment:
Let's use a remix form here.

Let's also make note in the comment that the key prop is what makes this work. 

Also, could you update the README to explain the example a bit and tell where the relevant bits of code are?

Reply:
hey @kentcdodds, have updated the README here https://github.com/remix-run/remix/pull/1121/commits/46823b189ca1a51c54b61a0ae4ffe70ebb02c52e. And have also added a note that the `key` prop does the magic of updating the Form with the new values here https://github.com/remix-run/remix/pull/1121/commits/a57543bb36fa7dd1f624b4d18e491462c84a74c2#diff-0db1e51405f843f23e57a234a98186c4009f1f66dd8b4af5543e0ac8cf508b76R44. Thanks!